### PR TITLE
docs: changes to create auth code api

### PR DESCRIPTION
**Pull request by AdithyaSSup**
Added flow diagram as well api spec changes tp`create Oauth2 auth code` api.

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -86,15 +86,17 @@ paths:
...
                       authorizationCode:
                         type: string
-                      timeCreated:
+                      timeCreatedMs:
                         $ref: '#/components/schemas/timestampMs'
-                      timeExpires:
+                      timeExpiresMs:
                         $ref: '#/components/schemas/timestampMs'
                   - type: object
                     properties:
```
Comment (rishabhpoddar) [inline api_spec.yaml:94]: flow diagram changes:
- do not implicitly add openid to scopes if the scopes array is empty. We should instead throw an error, but this is not yet finalised. Depends on https://github.com/supertokens/docs/pull/664#discussion_r1218187806

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 1 above)
Comment (AdithyaSSup) [reply api_spec.yaml:94]: for now i made it throw 400 error if scopes are empty in the flow diagram

Comment (AdithyaSSup) [reply api_spec.yaml:94]: Will accommodate, if there are any changes in adr later on.

File: api_spec.yaml [hunk 2]
```diff
@@ -86,15 +86,17 @@ paths:
...
                       status:
                         type: string
-                        enum: ['UNKNOWN_CLIENT_ID', 'INVALID_REDIRECT_URI', 'INVALID_SCOPES', 'UNKNOWN_CODE_CHALLENGE_METHOD', 'INVALID_ACCESS_TYPE', 'INVALID_CODE_CHALLENGE']
+                        enum: ['UNKNOWN_CLIENT_ID_ERROR',
+                        'UNKNOWN_SESSION_HANDLE_ERROR',
+                        'UNKNOWN_REDIRECT_URI_ERROR', 'UNKNOWN_SCOPE_ERROR', 'UNKNOWN_CODE_CHALLENGE_METHOD_ERROR', 'UNKNOWN_ACCESS_TYPE_ERROR', 'UNKNOWN_CODE_CHALLENGE_ERROR']
```
[the remark this answers is not in the document]
Comment (AdithyaSSup) [reply api_spec.yaml]: changes made

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**